PetGame/Services/PetService.cs

```diff
@@ -200,5 +221,267 @@ public bool DeletePet(ulong petId, ulong ownerId)
+        {
+            List<Activity> Activities = (List<Activity>)PastActivities;
+            //check to see if there were any activities in the specified interval
+            if (Activities.Count == 0)
+            {
+                HoursSinceLastActivity = HoursToCheck;
+            }
+            else
+            {
+                //get the last activity
+                Activity LastActivity = Activities[0];
+
+                //check the number of hours since the last activity
+                TimeSpan span = DateTime.Now.Subtract(LastActivity.Timestamp);
+                HoursSinceLastActivity = (int) span.TotalHours;
+            }
+            return HoursSinceLastActivity;
+        }
+
+        private int CalculateMinutesToNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
+        {
+            List<Activity> Activities = (List<Activity>)PastActivities;
+
+            if (Activities.Count == 0)
+            {
+                MinutesToNextAction = HoursToCheck * 60;
+            }
+            else
+            {
+                Activity LastActivity = Activities[0];
+
+                //check the number of minute since the last activity
+                int MinutesSinceLastActivity = 0;
+                TimeSpan span = DateTime.Now.Subtract(LastActivity.Timestamp);
+                MinutesSinceLastActivity = (int)span.TotalMinutes;
+
+                //if it's been more than 5 minutes, the user can perform another action
+                //otherwise, leave the default 5 min value in place
+                if (MinutesSinceLastActivity > 5)
+                {
+                    MinutesToNextAction = 0;
+                }
+            }
+            return MinutesToNextAction;
+        }
+
+        private double CalculateHunger(IEnumerable<Activity> PastActivities, double hungerPercentage, int HoursToCheck)
+        {
+            List<Activity> Activities = (List<Activity>)PastActivities;
+
+            if (Activities.Count == 0)
+            {
+                hungerPercentage -= (HungerDecreasePerHour * HoursToCheck);
+            }
+            else
+            {
+                //check the type of activity and subtract percentage accordingly
+                foreach (Activity Activity in PastActivities)
+                {
+                    //if a Feeding activity has occured, increase hunger
+                    if (Activity.Type.Equals(ActivityType.Feeding))
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, FeedingIncrease, PercentageOperation.Add);
+                    }
+                    //if a Training event has occured, subtract
+                    else if (Activity.Type.Equals(ActivityType.Training))
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, TrainingDecrease, PercentageOperation.Subtract);
```

Comment:
I still don't see why this is as complex as it is, couldn't you just get the same result by doing:
```cs
hungerPercentage = ConstrainPercentage(hungerPercentage - TrainingDecrease);
```

Reply:
Because I thought you wanted everything in separate functions

Reply:
I think you misunderstood what I meant. A function for `feed` that performs the logic on status when feeding, or a function for `train` that does the same was what I meant, so that in your `switch`/`case` you only call those methods, instead of doing all of the logic inline.

Reply:
Currently, you are repeating the same logic in more than one place.

Reply:
Gotcha. I think I see what you're getting at with this Modify function being too complex, I'll change it